hamming

whuan20011 opened the pull request:
Algorithm: dfs
watchout num in the recursive, num becomes 0, so we use a = num to track num, or it would be a dead loop. It shows exceed maxinum depth recursion
watchout the print style.

whuan20011 committed:
hamming
Algorithm: dfs
watchout num in the recursive, num becomes 0, so we use a = num to track num, or it would be a dead loop. It shows exceed maxinum depth recursion
watchout the print style.

smy20011 commented on hamming/hamming.py:
```diff
@@ -0,0 +1,70 @@
+            res.append(temp)
+            resres.append(str(a - 1))
+    recursive(N, B, D, a, res, resres)
+def hamming(D, temp, r):
```
It's better just return the diff and do the compare in the recursive code.

smy20011 commented on hamming/hamming.py:
```diff
@@ -0,0 +1,70 @@
+            b += 10
+        if q != 0:
+            print >>fout, " ".join(resres[(p * 10):])       
+def recursive(N, B, D, num, res, resres):
```
what is res and what is resres?

whuan20011 committed:
hamming
Algorithm: dfs(num + 1)
It is better return the diff from the hamming diff function to the dfs function

whuan20011 committed:
hamming
Algorithm: dfs
change: changed res, resres to meaningful names
